File: Exercice6/main.java

// Créez une classe ClientProfessionnel qui étend la classe Client et qui a comme attribut un numéro SIRET. 

class Main {
    public static void main(String[]args) {
        Client madina = new Client("Madina", 1234567);
        ProfessionalClient noah = new ProfessionalClient("Noah", 987456, 7412589);
        system.out.printIn(madina.name+" "+ madina.accountNumber);
        system.out.printIn(noah.name+" "+ noah.accountNumber+" "+noah.siretNumber);
    }
}

class Client {
//Variables
    String name;
    int accountNumber;

//constructor
    Client(String name, Int accountNumber){
        this.name = name;
        this.accountNumber = accountNumber;
    }
}

//Class
class ProfessionalClient extends Client {
    int siretNumber:

//Constructor
ProfessionalClient(String name, int accountNumber, int siretNumber){
  super(name, accountNumber);
  this.siretNumber = siretNumber;

    }
  
}

File: Exercice6/main.ts
// Créez une classe ClientProfessionnel qui étend la classe Client et qui a comme attribut un numéro SIRET. 

class Client {
    public name: string;
    public accountNumber: number;

    constructor(name: string, accountNumber: number) {
        this.name = name;
        this.accountNumber = accountNumber;
    }
}

class ClientProfessionnel extends Client {
    siretNumber: number;

    constructor(name: string, accountNumber: number, siretNumber: number) {
        super(name, accountNumber);
        this.siretNumber = siretNumber;
    }
}

let newClient: Client = new Client("Madina", 1234567899);
console.log(newClient.accountNumber, newClient.name);

let newClientPro: ClientProfessionnel = new ClientProfessionnel("Noah", 1234567889, 487615466);
console.log(newClientPro.accountNumber, newClientPro.name);